### PR TITLE
CMR-4650: Change JSON schema for DOI to support two different definitions

### PR DESCRIPTION
**Pull request by chris-durbin**
oneofus

This change allows the UMM for the DOI field to be either DOI and Authority or MissingReason and Explanation. We needed to add support for field level oneOf definitions in JSON schema in order to make this work.

### CODE REVIEW

File: umm-spec-lib/src/cmr/umm_spec/test/echo10_expected_conversion.clj [hunk 1]
```diff
@@ -259,6 +259,13 @@
       [(cmn/map->ResourceCitationType
         {:OtherCitationDetails other-citation-details})])))
 
+(defn- expected-doi
+  "Updates the expected DOI for ECHO10."
+  [doi]
+  (let [updated-doi (util/remove-nil-keys (dissoc doi :MissingReason :Explanation))]
```
Comment (mschmele) [inline umm-spec-lib/src/cmr/umm_spec/test/echo10_expected_conversion.clj]: Nitpick: Would a `when-let` work here? I see this `let -> when -> seq` in a few places

File: umm-spec-lib/src/cmr/umm_spec/xml_to_umm_mappings/iso_shared/doi.clj [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  "gov.nasa.esdis.umm.doi")
+
+(defn- is-doi-field?
+  "Returns true if the given gmd-id is for a DOI field."
+  [gmd-id]
+  (and (= (value-of gmd-id "gmd:MD_Identifier/gmd:description/gco:CharacterString") "DOI")
+       (= (value-of gmd-id "gmd:MD_Identifier/gmd:codeSpace/gco:CharacterString") doi-namespace)))
+
+(defn parse-doi
+  "There could be multiple CI_Citations. Each CI_Citation could contain multiple gmd:identifiers.
```
Comment (mschmele) [inline umm-spec-lib/src/cmr/umm_spec/xml_to_umm_mappings/iso_shared/doi.clj]: Super helpful docstring 👍 

Might be worth adding a sentence about why we're picking the first DOI for now

**Comment by eereiter** [comment]
I didn't comment on the clojure style stuff, just looked at the basic logic of the code and it looks good.

File: umm-spec-lib/src/cmr/umm_spec/test/echo10_expected_conversion.clj (hunk 1 above)
Comment (chris-durbin) [reply umm-spec-lib/src/cmr/umm_spec/test/echo10_expected_conversion.clj]: `updated-doi` is a map, not a collection. I was checking if it was empty (had no keys) with the `seq`. I don't actually want to return the seq of a map, so I couldn't use a `when-let`.

File: umm-spec-lib/src/cmr/umm_spec/xml_to_umm_mappings/iso_shared/doi.clj (hunk 2 above)
Comment (chris-durbin) [reply umm-spec-lib/src/cmr/umm_spec/xml_to_umm_mappings/iso_shared/doi.clj]: I can't take credit for this docstring. I don't think there's anything special about the first, that's just the pattern used when there are multiple of something in one format, but we can only use one in another format.

**Pull request merged**